### PR TITLE
Reference - Permission boundaries

### PR DESCRIPTION
**Pull request by offlineprogrammer**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] Swift
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/reference/permissions-boundary/index.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+
+To set the maximum permissions that can be granted to IAM Roles created by Amplify, configure a [permissions boundary](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) for the project. Then, Amplify-generated IAM roles can perform only the actions that are allowed by both the roles’ policies and permissions boundary. You can configure a different permissions boundary for each environment. For example, this enables you to deny a dev environment all access to a prod environment's resources.
```
Comment (josefaidt) [inline src/pages/[platform]/reference/permissions-boundary/index.mdx:32]:
```suggestion
To set the maximum permissions that can be granted to IAM Roles created by Amplify, configure a [permissions boundary](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) for the project. Then, Amplify-generated IAM roles can perform only the actions that are allowed by both the roles’ policies and permissions boundary. You can configure a different permissions boundary for each environment. For example, this enables you to deny a staging or dev environment access to a production environment's resources.
```

File: src/pages/[platform]/reference/permissions-boundary/index.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The IAM permissions boundary will apply to all IAM Roles created by Amplify. This includes the "auth role" assumed by users that log into the app and the "unauth role" assumed by guest users. It also applies to Lambda execution roles, Cognito user group roles, and any role configured in a [custom resource stack](/gen1/[platform]/tools/cli/custom/cloudformation/).
```
Comment (josefaidt) [inline src/pages/[platform]/reference/permissions-boundary/index.mdx:34]: nit for gen1 link

Comment (josefaidt) [reply src/pages/[platform]/reference/permissions-boundary/index.mdx:34]: can we link to
```
/[platform]/build-a-backend/add-aws-services/
```

File: src/pages/[platform]/reference/permissions-boundary/index.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The IAM Policy, to be used as a permissions boundary, must be configured outside of Amplify. A permissions boundary is an IAM Policy and can be created following the guide [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create-console.html). This is usually part of an AWS Organization rule or other corporate governance requirement. Once you have created an IAM Policy to use as a permissions boundary, copy the IAM Policy ARN for the next steps.
```
Comment (josefaidt) [inline src/pages/[platform]/reference/permissions-boundary/index.mdx:36]:
```suggestion
The IAM Policy to be used as a permissions boundary must be configured outside of Amplify. A permissions boundary is an [IAM Policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create-console.html). This is usually part of an AWS Organization rule or some other corporate governance requirement. Once you have created an IAM Policy to use as a permissions boundary, copy the IAM Policy ARN for the next steps.
```

File: src/pages/[platform]/reference/permissions-boundary/index.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Set up a permissions boundary in a new project
+
+To initialize a project with a permissions boundary run:
+
+```bash showLineNumbers={false}
```
Comment (josefaidt) [inline src/pages/[platform]/reference/permissions-boundary/index.mdx:42]:
````suggestion
```bash title="Terminal" showLineNumbers={false}
````

File: src/pages/[platform]/reference/permissions-boundary/index.mdx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+cdk bootstrap --custom-permissions-boundary <iam-policy-arn>
+```
+
+
+The `cdk bootstrap` command is a one-time operation that configures the AWS environment (account + region) for CDK deployments. Once executed, users can continue to utilize Amplify commands (e.g. sandbox) without interruption. Any custom IAM permissions boundary set by cdk bootstrap will be automatically applied to the roles created by Amplify.
```
Comment (josefaidt) [inline src/pages/[platform]/reference/permissions-boundary/index.mdx:47]:
```suggestion
The `cdk bootstrap` command is a one-time operation that configures the AWS account and region for CDK deployments. Once executed, users can continue to utilize Amplify commands (e.g. `sandbox`) without interruption. Any custom IAM permissions boundary set by `cdk bootstrap` will be automatically applied to the roles created by Amplify.
```

**Review by josefaidt — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Comment by srquinn21** [comment]
I know I'm late to the game here, but the part about being able to separate IAM permission boundaries between dev and prod environments via `cdk bootstrap` is not true. `cdk bootstrap  --custom-permissions-boundary` will apply globally to any CDK deployment in the account/region in which the bootstrap stack was deployed. Controlling this on a per environment basis will require passing cdk context or exposing the `Stage` or `App` construct props to the customer. See here for relevant documentation: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam-readme.html#bootstrap-permissions-boundary. A possible solution for customers that want this on a per Amplify branch basis will need to use a CDK `Aspect` to visit all Role nodes.

**Pull request merged**